### PR TITLE
Add inlineCss trade-offs/gotchas and remove canary check

### PR DESCRIPTION
**Pull request by gaojude**
- Add trade offs + gotchas for inlineCss doc
- Remove canary check for inlineCss

### CODE REVIEW

File: docs/01-app/03-api-reference/05-config/01-next-config-js/inlineCss.mdx [hunk 1]
```diff
@@ -4,6 +4,8 @@ description: Enable inline CSS support.
 version: experimental
 ---
 
+# Usage
```
Comment (huozhi) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/inlineCss.mdx:7]: might be too large to use h1, I think it's common to have h2 for those sections.

**Review by huozhi — COMMENTED** [review]
would be nice to have a PR description

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 3690cd060aa93fb0d95dd342ca6b07141c0b36a6

`pnpm test test/integration/amp-export-validation/test/index.test.js`
- AMP Validation on Export > production mode > should have shown errors during build

<details>
<summary>Expand output</summary>

  ● AMP Validation on Export › production mode › should have shown errors during build

    expect(received).toMatch(expected)

    Expected pattern: /error.*The mandatory attribute 'height' is missing in tag 'amp-video'\./
    Received string:  "   Loading config from /root/actions-runner/_work/next.js/next.js/test/integration/amp-export-validation/next.config.js
       Loading config from /root/actions-runner/_work/next.js/next.js/test/integration/amp-export-validation/next.config.js
       ▲ Next.js 15.0.4-canary.45·
       Checking validity of types ...
       Creating an optimized production build ...
       Loading config from /root/actions-runner/_work/next.js/next.js/test/integration/amp-export-validation/next.config.js
       Loading config from /root/actions-runner/_work/next.js/next.js/test/integration/amp-export-validation/next.config.js
       Loading config from /root/actions-runner/_work/next.js/next.js/test/integration/amp-export-validation/next.config.js
     ✓ Compiled successfully
       Collecting page data ...
       Generating static pages (0/8) ...
       Generating static pages (2/8)··
     ⚠ Linting is disabled.
    Error occurred prerendering page \"/first\". Read more: https://nextjs.org/docs/messages/prerender-error
    AssertionError: Assertion failed: WebAssembly is uninitialized

      at new module$contents$goog$asserts_AssertionError (../evalmachine.<anonymous>:102:1441)
      at module$contents$goog$asserts_doAssertFailure (../evalmachine.<anonymous>:103:354)
      at goog.asserts.assertExists (../evalmachine.<anonymous>:104:374)
      at Object.module$contents$amp$validator_validateString [as validateString] (../evalmachine.<anonymous>:2238:108)
      at Validator.validateString (../packages/next/dist/compiled/amphtml-validator/index.js:1:20650)
      at validateAmp (../packages/next/dist/export/routes/pages.js:100:34)
      at async exportPagesPage (../packages/next/dist/export/routes/pages.js:117:13)
      at async Span.traceAsyncFn (../packages/next/dist/trace/trace.js:153:20)
      at async exportPage (../packages/next/dist/export/worker.js:334:18)
      Export encountered an error on /first, exiting the build.
       ⨯ Static worker exited with code: 1 and signal: null
      "
      at Object.toMatch (integration/amp-export-validation/test/index.test.js:28:29)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: docs/01-app/03-api-reference/05-config/01-next-config-js/inlineCss.mdx (hunk 1 above)
Comment (samcx) [reply docs/01-app/03-api-reference/05-config/01-next-config-js/inlineCss.mdx:7]: Agreed!

**Review by samcx — CHANGES_REQUESTED** [review]

File: docs/01-app/03-api-reference/05-config/01-next-config-js/inlineCss.mdx [hunk 2]
```diff
@@ -28,3 +30,37 @@ const nextConfig = {
...
+While inlining CSS offers significant benefits for performance, there are scenarios where it may not be the best choice:
+
+- **Large CSS Bundles**: If your CSS bundle is too large, inlining it may significantly increase the size of the HTML, resulting in slower Time to First Byte (TTFB) and potentially worse performance for users with slow connections.
+- **Dynamic or Page-Specific CSS**: For applications with highly dynamic styles or pages that use different sets of CSS, inlining may lead to redundancy and bloat, as the full CSS for all pages may need to be inlined repeatedly.
+
+- **Browser Caching**: In cases where visitors frequently return to your site, external CSS files allow browsers to cache styles efficiently, reducing data transfer for subsequent visits. Inlining CSS eliminates this benefit.
+
+Evaluate these trade-offs carefully, and consider combining inlining with other strategies, such as critical CSS extraction or a hybrid approach, for the best results tailored to your site's needs.
+
+# Experimental Warning
```
Comment (samcx) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/inlineCss.mdx:59]: This should probably be a **Good to know: ** section.

**Review by huozhi — APPROVED** [review]

**Review by samcx — APPROVED** [review]

**Pull request merged**